daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/Runner.scala

```diff
@@ -203,6 +203,10 @@ object Script {
         for {
           scriptIds <- getScriptIds(result)
         } yield Script.Function(scriptExpr, param, scriptIds)
+      case GenDValue(TForall(_, ty), _, _) =>
+        for {
+          scriptIds <- getScriptIds(ty)
+        } yield Script.Action(scriptExpr, scriptIds)
```

Comment:
I will have drop all the quantifications not only one. 
And you still have to check that the result is a function.

Reply:
There is not reason that for instance `\/ a \/ b. Script (a, b)` and `\/a, Text -> Script a` do not work

Reply:
Do we want to support that though? I think only in the case where its `forall a. Script a` can we be sure that its safe to drop quant.
Imagine we had `forall a. a -> Script ()`, we couldn't possibly run this

Reply:
> There is not reason that for instance `∀ a . ∀ b. Script (a, b)` and `∀a . Text -> Script a` do not work

I agree those could work, but there are other cases that won't, we'd need to ensure the quantified types do not appear in inputs

Reply:
Though, I think I'm missing some context about the Function case, when is that used?

Reply:
I am fine with the restriction that imposes there is no free variables , but I  do not see why we should accept `∀ a . Script a` but not the other case I write above. 

In my point of view, either we allow free variables in a consistent way, either we do not at all.

Reply:
regarding the function case, I know there was a way to run `X -> Scenario Y` by requiring a JSON blob which is interpreted as an `X`, but I don't remember how this was accessed or if we provide the same functionality for `X -> Script Y`

Reply:
I think it should be possible to drop as many `forall`s as possible from `t` to get `t'`, then if `t'` is a function we check there are no type variables in the argument and that the result is a `Script _` (possibly with more `forall`s e.g. `Text -> ∀ a. Script a`), otherwise we just check that `t'` is a `Script _`

Reply:
Do we have nice ways to check if there are no type variables in an expression? Otherwise we may need to fully iterate the tree of each argument to impose this restriction.
as something like `forall a. (Either (a, Text) Text) -> Script ()` would not be acceptable
If this is what we want though, I'll will ofc implement it

Reply:
Ah, I didn't see Moises' comments before mine went through

Reply:
Sure, I can implement that. I'll note that the IDE does not detect scripts with arguments

Reply:
> Do we have nice ways to check if there are no type variables in an expression
We have `com.daml.lf.validation.TypeSubst#freeVars` in `//daml-lf/validation/`
I think am fine to either import the module in daml script or to move the logic in ``//daml-lf/language/`

>  I'll note that the IDE does not detect scripts with arguments
This can be called only from command line.